src/artemis/experiment_plans/tests/test_optimise_attenuation_plan.py

```diff
@@ -0,0 +1,101 @@
+from unittest.mock import MagicMock
+
+import numpy as np
+from bluesky.run_engine import RunEngine
+from dodal.beamlines import i03
+from dodal.devices.xspress3_mini.xspress3_mini import DetectorState
+from ophyd.status import Status
+
+from artemis.experiment_plans.optimise_attenuation_plan import (
+    PlaceholderParams,
+    optimise_attenuation_plan,
+)
+from artemis.parameters.beamline_parameters import get_beamline_parameters
+
+
+def fake_create_devices():
+    zebra = i03.zebra(fake_with_ophyd_sim=True)
+    zebra.wait_for_connection()
+    xspress3mini = i03.xspress3mini(fake_with_ophyd_sim=True)
+    xspress3mini.wait_for_connection()
+    attenuator = i03.attenuator(fake_with_ophyd_sim=True)
+    attenuator.wait_for_connection()
+    return zebra, xspress3mini, attenuator
+
+
+"""Default params:
+default_low_roi = 100
+default_high_roi = 2048
+increment = 2
+lower_lim = 20000
+upper_lim = 50000
+transmission = 0.1
+
+Produce an array with 1000 values between 0-1 * (1+transmission)
+"""
+CALCULATED_VALUE = 0
+
+
+def get_good_status():
+    status = Status()
+    status.set_finished()
+    return status
+
+
+def test_optimise(RE: RunEngine):
```

Comment:
Should: This test is good but we should have lots more that test small bits of behaviour (this might go along with cutting the bigger plan up) e.g.:
* Move the checking the `total_counts` bit out into a function, now write tests that this function gives True/False when expected
* Some test that after more than max_cycles we get an exception
* A test on just the arming section happening